src/client/sandbox/event/selection.ts

```diff
@@ -1,17 +1,22 @@
+/* eslint-disable no-unused-vars*/
+import TimersSandbox from '../timers';
+import EventSimulator from './simulator';
+/* eslint-enable no-unused-vars*/
+
 import FocusBlurSandbox from './focus-blur';
 import Listeners from './listeners';
 import nativeMethods from '../native-methods';
 import * as browserUtils from '../../utils/browser';
 import * as domUtils from '../../utils/dom';
 import INTERNAL_PROPS from '../../../processing/dom/internal-properties';
 
-const browserResetInputSelection = browserUtils.isFirefox && browserUtils.version > 50;
+const browserResetInputSelection = browserUtils.isFirefox;
 
 export default class Selection {
-    focusBlurSandbox: any;
-    timersSandbox: any;
-    listeners: any;
-    eventSimulator: any;
+    focusBlurSandbox: FocusBlurSandbox;
```

Comment:
We also have `_isNumberOrEmailInput` in Selection that now used only in `_needChangeInputType`. I suggest moving `_isNumberOrEmailInput` in dom utils and use it in `isInputWithoutSelectionProperties` and in `_needForInternalSelection`.